src/d3a/models/area/__init__.py

```diff
@@ -425,6 +427,12 @@ def _match_bids_offers(self) -> None:
                 bid_offer_pairs = bid_offer_matcher.get_matches_recommendations(data)
                 if not bid_offer_pairs:
                     break
+                if hasattr(bid_offer_matcher.match_algorithm, "state"):
+                    self.state.cumulative_offers[self.now] = (bid_offer_matcher.match_algorithm.
+                                                              state.cumulative_offers)
+                    self.state.cumulative_bids[self.now] = (bid_offer_matcher.match_algorithm.
+                                                            state.cumulative_bids)
+
```

Comment:
I would be in favour of refactoring this into a separate function called for example: `copy_clearing_state_to_area_state` (there can only be a better name ;-) )